lock-api/src/main/java/com/palantir/lock/client/RemoteTimelockServiceAdapter.java

```diff
@@ -44,11 +45,12 @@ private RemoteTimelockServiceAdapter(
             NamespacedTimelockRpcClient rpcClient,
             NamespacedConjureTimelockService conjureTimelockService,
             LockWatchEventCache lockWatchEventCache,
-            LeaderTimeGetter leaderTimeGetter) {
+            LeaderTimeGetter leaderTimeGetter,
+            Function<LockLeaseService, IdentifiedAtlasDbTransactionStarter> batchingTransactionStarterFactory) {
         this.rpcClient = rpcClient;
         this.lockLeaseService = LockLeaseService.create(conjureTimelockService, leaderTimeGetter);
-        this.transactionStarter = TransactionStarter.create(
-                lockLeaseService, StartTransactionsLockWatchEventCache.create(lockWatchEventCache));
+        this.transactionStarter =
+                TransactionStarter.create(lockLeaseService, batchingTransactionStarterFactory.apply(lockLeaseService));
```

Comment:
nit: Unsure if this is preferable, but it might be better to just pass in the Factory here, to ensure that the lockLeaseService used in the TransactionStarter (whether directly or indirectly) is the same one.